src/main/java/de/tum/in/www1/artemis/service/plagiarism/ModelingPlagiarismDetectionService.java

```diff
@@ -55,6 +65,8 @@ public ModelingPlagiarismResult checkPlagiarism(ModelingExercise exerciseWithPar
 
         result.setExercise(exerciseWithParticipationsSubmissionsResults);
 
+        plagiarismCacheService.disablePlagiarismCheck(courseId);
```

Comment:
we need to make sure that this is always invoked, even if an exception occurs, so please put the code above in a try catch statement, re-throw the exception and invoked `disablePlagiarismCheck` in the final clause